repl/format.go

```diff
@@ -1,116 +1,142 @@
+/*
+
+Copyright (c) 2019 Martin Atkins
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+*/
+
 package repl
 
 import (
-	"bufio"
-	"bytes"
 	"fmt"
-	"sort"
 	"strconv"
 	"strings"
-)
 
-// FormatResult formats the given result value for human-readable output.
-//
-// The value must currently be a string, list, map, and any nested values
-// with those same types.
-func FormatResult(value interface{}) (string, error) {
-	return formatResult(value, false)
-}
+	"github.com/zclconf/go-cty/cty"
+)
 
-func formatResult(value interface{}, nested bool) (string, error) {
-	if value == nil {
-		return "null", nil
+// FormatValue formats a value in a way that resembles Terraform language syntax
+// and uses the type conversion functions where necessary to indicate exactly
+// what type it is given, so that equality test failures can be quickly
+// understood.
+func FormatValue(v cty.Value, indent int) string {
+	if !v.IsKnown() {
+		return "(unknown)"
```

Comment:
I'd suggest changing this to `(known after apply)` to match with the `terraform plan` output. We've seen in the past that there are unfortunately a few other edge-cases where things appear as unknown in the console that are _not_ caused by something that would be set after apply, but I believe we addressed the most common of those recently by making an exception to allow the impure (and thus, non-convergent) functions like `timestamp` to be evaluated in the console.